src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
Talked to @spadapet offline: if there are no specific requirements for deterministic ordering, we can simply order by type name, as I added in commit https://github.com/NuGet/NuGet.Client/pull/664/commits/861bd338b69ddee4aeec00fdfef39e4c231f6fa3